src/__tests__/indexing.ts

```diff
@@ -0,0 +1,292 @@
+/**
+ * @jest-environment ./build/index.js
+ */
+import { jest } from '@jest/globals'
+import { CeramicApi, Page, StreamState, StreamUtils, TestUtils } from '@ceramicnetwork/common'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { StreamID } from '@ceramicnetwork/streamid'
+import { config } from 'node-config-ts'
+import { createDid } from '../utils.js'
+import { DID } from 'dids'
+
+const TEST_MODEL = StreamID.fromString(config.jest.models[0])
+
+declare global {
+  const ceramic: CeramicApi
+  const ceramicClient: CeramicApi
+}
+
+const extractStreamStates = (page: Page<StreamState | null>): Array<StreamState> => {
+  if (page.edges.find(edge => edge.node === null)) {
+    throw new Error(
+      'null stream state found. This may indicate a problem with data persistence of your state store, which can result in data loss.'
+    )
+  }
+
+  return page.edges.map(edge => edge.node) as Array<StreamState>
+}
+
+const extractDocuments = (
+  ceramic: CeramicApi,
+  page: Page<StreamState | null>
+): Array<ModelInstanceDocument> => {
+  return extractStreamStates(page).map(state =>
+    ceramic.buildStreamFromState<ModelInstanceDocument>(state)
+  )
+}
+
+describe('indexing', () => {
+  describe('Using existing model', () => {
+    jest.setTimeout(1000 * 60)
+    const originalDid = ceramic.did as DID
+
+    const singleNodeTestCases: any[] = []
+    const twoNodesTestCases: any[] = []
+
+    if (config.jest.services.ceramic.indexingEnabled) {
+      singleNodeTestCases.push(['ceramic', ceramic])
+    }
+
+    if (config.jest.services.ceramicClient.indexingEnabled) {
+      singleNodeTestCases.push(['ceramicClient', ceramicClient])
+    }
+
+    if (
+      config.jest.services.ceramic.indexingEnabled &&
+      config.jest.services.ceramicClient.indexingEnabled
+    ) {
+      twoNodesTestCases.push(['creating: ceramic, loading: ceramicClient', ceramic, ceramicClient])
+      twoNodesTestCases.push(['creating: ceramicClient, loading: ceramic', ceramicClient, ceramic])
+    }
+
+    afterEach(() => {
+      ceramic.did = originalDid
+      ceramicClient.did = originalDid
+    })
+
+    if (singleNodeTestCases.length > 0) {
+      test.each(singleNodeTestCases)(
+        'Can create and query on same node -- %s',
+        async (_, ceramicInstance: CeramicApi) => {
+          const doc1 = await ModelInstanceDocument.create(
+            ceramicInstance,
+            { data: Math.floor(Math.random() * 1000) },
```

Comment:
any reason this needs to be random?  When I read something like this without context, I assume there must be a reason so I start looking for what it is.  I think as a general rule when writing tests we should aim to keep things as simple and straightforward as possible so it's clear which parts actually matter to the core of what's being tested.

As far as I can tell this test would work just as well if all the data in all the documents was always fixed and the same in each test run